### PR TITLE
Fields api

### PR DESCRIPTION
**Pull request by yashashkumar**

### CODE REVIEW

File: api-service/src/controllers/DatasetList/DatasetList.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export const apiId = "api.datasets.list"
 export const errorCode = "DATASET_LIST_FAILURE"
 const liveDatasetStatus = ["Live", "Retired", "Purged"]
 const draftDatasetStatus = ["Draft", "ReadyToPublish"]
-const defaultFields = ["dataset_id", "name", "type", "status", "tags", "version", "api_version", "dataset_config", "created_date", "updated_date"]
+const defaultFields = ["dataset_id", "name", "type", "status", "tags", "version", "api_version", "dataset_config", "created_date", "updated_date","data_schema"]
```
Comment (SanthoshVasabhaktula) [inline api-service/src/controllers/DatasetList/DatasetList.ts:15]: Why do we need data_schema as the default fields?

File: api-service/src/routes/Router.ts [hunk 2]
```diff
@@ -54,6 +55,8 @@ router.post("/datasets/copy", setDataToRequestObject("api.datasets.copy"), onReq
 router.post("/connectors/list", setDataToRequestObject("api.connectors.list"), onRequest({ entity: Entity.Management }), ConnectorsList);
 router.get("/connectors/read/:id", setDataToRequestObject("api.connectors.read"), onRequest({entity: Entity.Management }), ConnectorsRead);
 router.post("/datasets/import", setDataToRequestObject("api.datasets.import"), onRequest({ entity: Entity.Management }), DatasetImport);
+router.get("/datasets/generate-fields/:dataset_id", setDataToRequestObject("api.generate.fields"), onRequest({ entity: Entity.Management }), generateFields);
```
Comment (SanthoshVasabhaktula) [inline api-service/src/routes/Router.ts:58]: shouldn't this be a webconsole API? Move it to webconsole with the path as /webconsole/dataset/flatten/:datasetId

**Review by SanthoshVasabhaktula — CHANGES_REQUESTED** [review]

File: api-service/src/routes/Router.ts (hunk 2 above)
Comment (yashashkumar) [reply api-service/src/routes/Router.ts:58]: done

File: api-service/src/controllers/DatasetList/DatasetList.ts (hunk 1 above)
Comment (yashashkumar) [reply api-service/src/controllers/DatasetList/DatasetList.ts:15]: currently im using it in web-console 
i'll remove after implementing the fields api

**Pull request closed without merging**